### PR TITLE
DALA-2767 Next Generation Dev-Tools

### PR DESCRIPTION
**Pull request by MarcTM01**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] The version of main currently active on prod is deployed to a dev server with `Reset non-user related Docker Volumes & Re-populate` turned on
  - [ ] It's verified that the CD run is green
  - [ ] The data from prod is migrated using `./migrateData.sh dataland.com <SOURCE_API_KEY> <TARGET> <TARGET_API_KEY>` 
  - [ ] The feature branch is deployed to the same server with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the dev server
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server
- [x] Make sure that pull requests in the development tools repository, which are connected to this pull requests, 
are merged

### CODE REVIEW

File: dataland-frontend/tests/e2e/fixtures/frameworks/sfdr/index.ts [hunk 1]
```diff
@@ -23,3 +23,5 @@ export function exportFixturesSfdrData(): void {
     JSON.stringify(preparedFixtureSfdrData, null, "\t"),
   );
 }
+
+export default exportFixturesSfdrData;
```
Comment (SiegfriedSobkowiak) [inline dataland-frontend/tests/e2e/fixtures/frameworks/sfdr/index.ts:27]: hier auch

File: dataland-frontend/tests/e2e/fixtures/frameworks/p2p/index.ts [hunk 2]
```diff
@@ -19,3 +19,5 @@ export function exportFixturesP2p(): void {
     JSON.stringify(preparedFixtureP2pDataForFinancials, null, "\t"),
   );
 }
+
+export default exportFixturesP2p;
```
Comment (SiegfriedSobkowiak) [inline dataland-frontend/tests/e2e/fixtures/frameworks/p2p/index.ts:23]: und nochmal

File: dataland-framework-toolbox/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    implementation(libs.log4j.slf4j2.impl)
+    implementation(libs.spring.context)
+    implementation(libs.spring.test)
+    implementation(libs.jackson.dataformat.csv)
+    implementation(libs.jackson.module.kotlin)
+    implementation(libs.commons.io)
+    implementation(libs.commons.codec)
+    implementation(libs.commons.text)
+    implementation(libs.slf4j.api)
+    implementation("org.gradle:gradle-tooling-api:7.1.1")
```
[the remark this answers is not in the document]
Comment (MarcTM01) [reply dataland-framework-toolbox/build.gradle.kts]: Fixed

File: dataland-framework-toolbox/scripts/convert-excel-to-csv.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+# BLANK - No special column formatters
+# 1033 - Use "English - United States" Formatting
+# false - Quoted field as text --> NO (default)
+# true - Detect special numbers (default)
+# false - Save cell contents as shown --> No
+# false - Export cell formulas --> No
+# false - Remove spaces --> No
+# -1 - Export all sheets
+
+libreoffice --headless \
```
[the remark this answers is not in the document]
Comment (MarcTM01) [reply dataland-framework-toolbox/scripts/convert-excel-to-csv.sh:22]: Fixed

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/InDevelopmentPavedRoadFramework.kt [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        // Register custom converter that in this case just ignores all unknown fields.
+        val noopComponentFactory = object : TemplateComponentFactory {
+            override fun canGenerateComponent(row: TemplateRow): Boolean = true
+
+            override fun generateComponent(
+                row: TemplateRow,
+                utils: ComponentGenerationUtils,
+                componentGroup: ComponentGroupApi,
+            ): ComponentBase? {
+                println("No-one wants to generate components for ${row.component}")
```
[the remark this answers is not in the document]
Comment (MarcTM01) [reply dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/InDevelopmentPavedRoadFramework.kt]: Fixed

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/utils/DatalandRepositoryGradleInterface.kt [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        val gradleConnector = GradleConnector.newConnector()
+        val projectConnection = gradleConnector
+            .forProjectDirectory(repository.path.toFile())
+
+        projectConnection.connect().use(usingFunction)
+    }
+
+    /**
+     * Execute a list of gradle tasks
+     * @param force if true, fore re-execution even if the task is up-to-date
```
[the remark this answers is not in the document]
Comment (MarcTM01) [reply dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/utils/DatalandRepositoryGradleInterface.kt]: Fixed

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/utils/Naming.kt [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * A utility for converting between naming conventions
+ */
+object Naming {
+    /**
+     * Converts an input label (in "Sentence case") to camelCase (e.g., "sentenceCase")
+     */
+    fun getNameFromLabel(label: String, useCapitalCase: Boolean = false): String {
+        val sanitizedLabel = label.replace("&", " And ")
+        return CaseUtils.toCamelCase(sanitizedLabel, useCapitalCase, '-', ',', '/', '(', ')', ':')
```
[the remark this answers is not in the document]
Comment (MarcTM01) [reply dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/utils/Naming.kt]: Added

File: dataland-frontend/package.json [hunk 8]
```diff
@@ -8,6 +8,7 @@
     "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix",
     "lintci": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --max-warnings 0 --no-fix",
     "cypress": "cypress open",
+    "checkfrontendcompilation": "npx tsc --noEmit -P tsconfig.app.json",
```
[the remark this answers is not in the document]
Comment (MarcTM01) [reply dataland-frontend/package.json]: Removed

**Pull request merged**